spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/SecretManagerPropertySourceLocator.java

```diff
@@ -60,11 +61,22 @@ public void setVersions(Map<String, String> versions) {
 
 	@Override
 	public PropertySource<?> locate(Environment environment) {
-		return new SecretManagerPropertySource(
-				SECRET_MANAGER_NAME,
-				this.client,
-				this.projectIdProvider,
-				this.secretsPrefix,
-				this.versions);
+		CompositePropertySource compositePropertySource = new CompositePropertySource(SECRET_MANAGER_NAME);
+
+		compositePropertySource.addPropertySource(
```

Comment:
Do we want to have both resolvable? The new version is more comprehensive. Or are we doing it for backwards compatibility?

Perhaps we could automatically configure only the newest property source, and provide a "legacy" property for  loading the old one?

Reply:
Yeah I wasn't sure how to handle this. I think would prefer if we only have the new version. Maybe if it's ok, I'd like to just remove the existing functionality and replace it with the new method. @meltsufin 

I guess it doesn't make sense for both to co-exist since the legacy one already does the work of downloading all the secrets, and then if the new system exists they might redownload secrets they already loaded into their properties.